cli/pcluster/config/validators.py

```diff
@@ -1120,3 +1120,34 @@ def fsx_ignored_parameters_validator(section_key, section_label, pcluster_config
                 errors.append(FSX_MESSAGES["errors"]["ignored_param_with_fsx_fs_id"].format(fsx_param=fsx_param))
 
     return errors, warnings
+
+
+def ebs_volume_type_size_validator(section_key, section_label, pcluster_config):
+    """
+    Validate that the EBS volume size matches the chosen volume type.
+
+    The default value of volume_size for EBS volumes is 20 GiB.
+    The volume size of gp2 ranges from 1 GiB - 16 TiB
+    The volume size of io1 ranges from 4 GiB - 16 TiB
+    The volume sizes of st1 and sc1 range from 500 GiB - 16 TiB
+    """
+    errors = []
+    warnings = []
+
+    section = pcluster_config.get_section("ebs", section_label)
+    volume_size = section.get_param_value("volume_size")
+    volume_type = section.get_param_value("volume_type")
+    volume_type_to_volume_size_bounds = {
+        "io1": (4, 16 * 1024),
+        "gp2": (1, 16 * 1024),
+        "st1": (500, 16 * 1024),
+        "sc1": (500, 16 * 1024),
+    }
+    min_size, max_size = volume_type_to_volume_size_bounds.get(volume_type)
```

Comment:
What happens when the `volume_type` isn't in the dict?

Reply:
Done! Added an if condition to check whether the volume_type is in the dict.